01week/datatypes.js

```diff
@@ -0,0 +1,186 @@
+// so I will expand the function to be more explicit in it's output
+//
+// first check the data types that are truthy
+// second check the data types that are falsy
+//  - this is bc some truthy types when checked will return falsy in functions like isNaN
+//  - ex: isNaN("this is a string") will return true, so should be placed AFTER the string check
+
+const checkDataType=(arg1)=>{
+  // begin checking TRUTHY data types
+  // check for Boolean type
+  if (typeof arg1 === "boolean") {
+    console.log(arg1 + " is type Boolean");
+  }
+  // check for Number type
+  // we need to check both typeof arg1 && isNaN(arg1) === "false" bc typeof NaN === "number"
+  else if (typeof arg1 === "number" && isNaN(arg1) === false) {
+    console.log(arg1 + " is type Number");
+  }
+  // check for String type
+  else if (typeof arg1 === "string") {
+    console.log(arg1 + " is type String");
+  }
+  // begin checking FALSY data types
+  // check for null type
+  else if (arg1 === null) {
+    console.log(arg1 + " is type Null");
+  }
+  // check for Undefinded type
+  else if (arg1 === undefined) {
+    console.log(arg1 + " is type Undefined");
+  }
+  // check for NaN type using isNaN method
+  else if (isNaN(arg1)) {
+    console.log(arg1 + " is type NaN");
+  }
+};
+
+// check Boolean type
+checkDataType(true);
+// check Null type
+checkDataType(null);
+// check Undefinded type
+checkDataType(undefined);
+// check Number type
+checkDataType(31);
+// check NaN type
+// parseInt("string not a number") will return a value of NaN
+checkDataType(parseInt("string not a number"));
+// check String type
+checkDataType("Howdydoodee");
+
+console.log(); // blank line separator
+
+// 5. write a js function that adds two numbers together
+//
+// oneline function that returns the addition of two arguments
+
+const addTwoNumbers=(arg1, arg2)=> arg1 + arg2;
+
+console.log(addTwoNumbers(20, 40));
+
+console.log(); // blank line separator
+
+// 6. write a js function that runs only when 2 'things' (args) are true
+//
+// the wording of the problem says runs only when 2 things are "true" and not "truthy"
+// so I assume the expected arguments are boolean type
+//
+// check both args given to see if they === true, if so, print message
+// if 1 or more args !== true, do nothing
+
+const twoArgsTrue=(arg1, arg2)=>{
+  if (arg1 === true && arg2 === true) {
+    console.log("Both args are true");
+  }
+};
+// this should print a message
+twoArgsTrue(true, true);
+// this should not print a message
+twoArgsTrue(true, false);
+
+console.log(); // blank line separator
+
+// 7. write a js function that runs when 1 of 2 things are true
+//
+// the wording of the problem says runs only when 1 of 2 things are "true" and not "truthy"
+// so I assume the expected arguments are boolean type
+//
+// check both args, if 1 arg === true && 1 arg === false, then print message
+// any other case, do not run
+
+const oneTrueOneFalse=(arg1, arg2)=>{
+  if (arg1 == true && arg2 === false) {
```

Comment:
This evaluates if the arguments are the boolean `true` and the boolean `false`.  I need a function that if one argument evaluates to true.